docs/usage.md

```diff
@@ -45,7 +45,7 @@ have the client automatically start when the Rest bot starts.
 
 ### Client lifetime management
 
-While Hikari's bots provide systems for stating and stopping sub-components,
+While Hikari's bots provide systems for starting and stopping sub-components,
```

Comment:
Add example of the listener type being explicitly passed to the callback plus unions